### ISSUE TITLE
running example of logistic error

### ISSUE THREAD

**Issue by Barry-ljf** [tf-encrypted/tf-encrypted#853]
when i run the demo of logistic 
i have do the instruction with the sitution of using ubuntu(not docker,just in conda):
![2022-03-02 10-00-21 的屏幕截图](https://user-images.githubusercontent.com/79430791/156280306-d8519d03-85c1-486c-84e4-9d288fc2ec01.png)
and the version of tensorflow is 1.15.2 as requirement. but  i cannot run the demo normally.
is warning of  version stop the process of this demo program? 
![2022-03-02 10-06-41 的屏幕截图](https://user-images.githubusercontent.com/79430791/156280937-b9aaa627-574c-4cf3-a3c8-2d892ecc7b41.png)

**Comment by vincehong** [comment]
Which script are you running and what is the error message ?

**Comment by Barry-ljf** [comment]
hi!
the code I have ran is  the vertical of logistic of example in Gemini-lab
the instruction is as follows:
 https://alibaba-gemini-lab.github.io/docs/blog/tfe/
I don't know why the program cannot run as expected,but there is no error information just stopped at this moment:
![2022-03-02 12-23-35 的屏幕截图](https://user-images.githubusercontent.com/79430791/156294712-a320739e-16f7-4d72-9a90-c9e9238227b0.png)

**Comment by Barry-ljf** [comment]
the config.json I have wrote is:
{
    "server0": "machine1:43198",
    "server1": "machine2:27919",
    "server2": "machine3:27921"
}
the port I have chosen is by inputting netstat -a  and find three random port is listening

**Comment by Barry-ljf** [comment]
by the way,  my directory set is as follows:
├── commom.py
├── CONFIG.json
├── machine1
│   ├── aliceTrainFile.csv
│   ├── common.py
│   ├── config.json
│   ├── __pycache__
│   │   └── common.cpython-36.pyc
│   └── training_alice.py
├── machine2
│   ├── bobLabel.csv
│   ├── config.json
│   └── training_bob.py
├── machine3
│   ├── config.json
│   └── training_server.py
├── training_alice.py
├── training_bob.py
└── training_server.py

**Comment by vincehong** [comment]
> hi! the code I have ran is the vertical of logistic of example in Gemini-lab the instruction is as follows: https://alibaba-gemini-lab.github.io/docs/blog/tfe/ I don't know why the program cannot run as expected,but there is no error information just stopped at this moment: ![2022-03-02 12-23-35 的屏幕截图](https://user-images.githubusercontent.com/79430791/156294712-a320739e-16f7-4d72-9a90-c9e9238227b0.png)

Your picture shows that server1 and server2 are halting and waiting for server0, this is normal. When you ran server0, the result will be printed on server0.

On very rare cases (I encountered), server0 will halt too, this may be because of bad network condition. Try choose other IP/ports, or use all the three workers on 127.0.0.1 to test if everything works.

**Comment by Barry-ljf** [comment]
Finally I made it!
Thanks！！
when I use tfe their own code , just one command windows can realize training jointly , so I want to learn how to realize a code in this way(like three directories). hh~
and very thanks for helping!

...
Batch  349
[[-0.02972108]
 [ 0.0405426 ]
 [-0.00335315]
...

**Comment by Barry-ljf** [comment]
By the way, Cheng ge,  I recently read faster secure adaptive Gradient Descent  by your team. is  the code of this essay  open source?

**Comment by vincehong** [comment]
> By the way, Cheng ge, I recently read faster secure adaptive Gradient Descent by your team. is the code of this essay open source?

Not yet, but MP-SPDZ proposed a improved version based on ours, and is open-sourced. See https://arxiv.org/abs/2107.00501?context=cs

**Comment by Barry-ljf** [comment]
thanks a lot ,I will go for it !
